### PR TITLE
Configure Reactive SQL Client (#18)

### LINKED ISSUE #18: `SpringDataHibernateMutinyReactiveQueryFactory` is not bean
When I ran the project, not found `SpringDataHibernateMutinyReactiveQueryFactory` bean

### PR DESCRIPTION
**Pull request by BAEKJungHo**
## Related Issue

- close #18   

## Description

- Configure Reactive SQL Client
- Refactor build.gralde.kt
- __[Warning] The application may run, but the bug may still be present.__
- __[Warning] Temporary code exists for testing. It will be deleted later.__

## References

- https://github.com/line/kotlin-jdsl/tree/main/spring/data-hibernate-reactive-jakarta
- https://github.com/line/kotlin-jdsl/blob/main/spring/data-reactive-core/README.md#quick-start---jpa-30
- https://vertx.io/docs/vertx-jdbc-client/java/

### CODE REVIEW

File: core/src/main/kotlin/team/backend/config/ReactiveQueryConfig.kt [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+            sessionFactory = sessionFactory,
+            subqueryCreator = subqueryCreator
+        )
+    }
+
+    private fun jpaVendorAdapters(): JpaVendorAdapter = HibernateJpaVendorAdapter().apply {
+        setDatabasePlatform("org.hibernate.dialect.MySQLDialect")
+    }
+
+    private fun jpaProperties(): Properties = Properties().apply {
```
Comment (mangchhe) [inline core/src/main/kotlin/team/backend/config/ReactiveQueryConfig.kt:95]: Doesn't it conflict with the yml file?

File: build.gradle.kts [hunk 2]
```diff
@@ -33,6 +33,28 @@ allprojects {
...
+	 */
+	implementation("org.springframework.boot:spring-boot-starter-data-jpa") {
+		exclude(module = "hibernate-core")
+		exclude(module = "hibernate-commons-annotations")
+	}
+	implementation("com.linecorp.kotlin-jdsl:spring-data-kotlin-jdsl-hibernate-reactive-jakarta:2.2.1.RELEASE")
+	implementation("org.hibernate.reactive:hibernate-reactive-core-jakarta:1.1.9.Final")
+	implementation("io.vertx:vertx-jdbc-client:4.3.1")
+	implementation("io.agroal:agroal-pool:2.0")
+	implementation("io.smallrye.reactive:mutiny-kotlin:1.6.0")
```
Comment (mangchhe) [inline build.gradle.kts:54]: 2.2.0 version is up to date, should I use 1.6.0?

File: build.gradle.kts [hunk 3]
```diff
@@ -33,6 +33,28 @@ allprojects {
...
+	 	Kotlin JDSL Examples
+	 	    https://github.com/cj848/kotlin-jdsl-example
+	 */
+	implementation("org.springframework.boot:spring-boot-starter-data-jpa") {
+		exclude(module = "hibernate-core")
+		exclude(module = "hibernate-commons-annotations")
+	}
+	implementation("com.linecorp.kotlin-jdsl:spring-data-kotlin-jdsl-hibernate-reactive-jakarta:2.2.1.RELEASE")
+	implementation("org.hibernate.reactive:hibernate-reactive-core-jakarta:1.1.9.Final")
+	implementation("io.vertx:vertx-jdbc-client:4.3.1")
```
Comment (mangchhe) [inline build.gradle.kts:52]: 4.4.1 version is up to date, should I use 4.3.1?

**Review by mangchhe — APPROVED** [review]
Good!

**Comment by mangchhe** [comment]
@BAEKJungHo 
I'm facing the error. Is there anything I need to configure separately?
```
HR000021: DDL command failed [org.hibernate.reactive.exception.VertxSqlClientException: could not execute statement]
```

File: core/src/main/kotlin/team/backend/config/ReactiveQueryConfig.kt (hunk 1 above)
Comment (BAEKJungHo) [reply core/src/main/kotlin/team/backend/config/ReactiveQueryConfig.kt:95]: Yes. yml doesn't requirement. I was deleted 

- https://github.com/withDogecoin/chaterview-private/pull/2

File: build.gradle.kts (hunk 3 above)
Comment (BAEKJungHo) [reply build.gradle.kts]: Updated 4.4.1

File: build.gradle.kts (hunk 2 above)
Comment (BAEKJungHo) [reply build.gradle.kts]: Updated 2.2.0

**Comment by BAEKJungHo** [comment]
> @BAEKJungHo I'm facing the error. Is there anything I need to configure separately?
> 
> ```
> HR000021: DDL command failed [org.hibernate.reactive.exception.VertxSqlClientException: could not execute statement]
> ```

I get the same warning. but there seems to be no problem in creating a table based on the entity.

**Pull request merged**